public/models/Player.js

```diff
@@ -12,39 +12,123 @@ class Player extends GameObject { // Marianna
     this.isDone = false;
   }
 
-  //Block types
-        //0 path
-        //1 wall
-        //2 goal
-        //3 player
-    //Marianna
-    //check for collision
-    //based on player rotation check only adjecent tiles in correct direction
-    //check based on tilemap
-    //for future: instead of collision might just return tiletype if there is collision and based on that act
-    isBlockCollision (map, direction, blockType, columnMovement, rowMovement){
-      let collision = false;
-      let x = this.x + columnMovement || this.x;
-      let y = this.y + rowMovement || this.y;
-      //row that the player is in
-      let row = Math.floor(y / map.tileHeight);
-      //column that the player is in
-      let column = Math.floor(x / map.tileWidth);
-      switch (direction) {
-          case "up" :
-              if ((y % map.tileHeight !== 0 && map.tiles[row - 1][column] === blockType) || (x % map.tileWidth!== 0 && map.tiles[row - 1][column + 1] === blockType)) collision = true;
-              break;
-          case "down" :
-              if ((y % map.tileHeight !== 0 && map.tiles[row][column] === blockType) || (x % map.tileWidth !== 0 && map.tiles[row][column + 1] === blockType)) collision = true;
-              break;
-          case "right" :
-              if ((x % map.tileWidth!== 0 && map.tiles[row-1][column + 1] === blockType) || (y % map.tileHeight !== 0 && map.tiles[row][column + 1] === blockType)) collision = true;
-              break;
-          case "left" :
-              if ((x % map.tileWidth!== 0 && map.tiles[row-1][column] === blockType) || (y % map.tileHeight !== 0 && map.tiles[row ][column] === blockType)) collision = true;
-              break;
-          default:
-      }
-      return collision;
+  // Block types
+  // 0 path
+  // 1 wall
+  // 2 goal
+  // 3 player
+  // Marianna
+  // check for collision
+  // based on player rotation check only adjecent tiles in correct direction
+  // check based on tilemap
+  // for future: instead of collision might just return tiletype if there is collision and based on that act
+  isBlockCollision(map, direction, columnMovement, rowMovement) {
+    let collision = false;
+    const x = this.x + columnMovement || this.x;
```

Comment:
what does it mean column movement?

Reply:
basically x movement and y movement if the player moves. It is for checking wall collisions so firstly it checks where the player would be after movement and if it is not colliding with wall, only then the player will move. But it would probably make more sense to rename it then to x and y movement? But then there are already a lot of x and y variables there. what do you think?